character.py

```diff
@@ -65,6 +65,46 @@ def add_minion(self, minion):
     def remove_minion(self, minion):
         self.minions.remove(minion)
 
+    def before_attack(self):
```

Comment:
This looks like I restored it in a bad merge, should all of these methods be deleted?

Reply:
Looks like these are copies of methods below, so we can delete them